bundles/core/src/main/java/com/adobe/cq/wcm/core/components/models/Download.java

```diff
@@ -0,0 +1,102 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2018 Adobe Systems Incorporated
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+
+package com.adobe.cq.wcm.core.components.models;
+
+import com.adobe.cq.export.json.ComponentExporter;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Sling model for the Download component.
+ */
+public interface Download extends ComponentExporter {
+
+    String PN_TITLE_FROM_ASSET = "titleFromAsset";
+    String PN_DESCRIPTION_FROM_ASSET = "descriptionFromAsset";
+
+
+    /**
+     * Returns either the title configured in the dialog or the title of the DAM asset,
+     * depending on the state of the titleFromAsset checkbox.
+     *
+     * @return the download title
+     */
+    default String getTitle(){
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Returns either the description configured in the dialog or the title of the DAM asset,
+     * depending on the state of the descriptionFromAsset checkbox.
+     *
+     * @return the download title
+     */
+    default String getDescription(){
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Returns the url to the asset.
+     *
+     * @return the asset url
+     */
+    default String getDownloadUrl(){
```

Comment:
I'd call it `getUrl()` to be consistent with other method names like `getTitle`, `getDescription`.